src/Product.java

```diff
@@ -0,0 +1,69 @@
+import java.util.*;
+
+public class Product {
+
+    private String productName;
+    private List<Map<String, String>> productRows;
+
+    public Product(String productName) {
+        this.productName = productName;
+        productRows = new ArrayList<>();
+    }
+
+    public String getProductName() {
+        return this.productName;
+    }
+
+    public void addProductRow(Map<String, String> productRow) {
+        productRows.add(productRow);
+    }
+
+    public List<String> calculateOutput(int startingYear, int lastYear) {
```

Comment:
Mixing the data objects with calculation logic is usually a bad idea. We ideally want to separate them in case the calculation logic or the data object changes. I'd take this out and put it in a mapper class that maps the Product to a Triangle